camera/src/Camera.cpp

```diff
@@ -0,0 +1,166 @@
+#include <camera/Camera.hpp>
+
+#include <data_types/math/math_constants.hpp>
+
+namespace rcbe::rendering {
+Camera::Camera(const rendering::RenderingContextPtr &context, const math::Vector3d &position, const math::Vector3d &lookat, const math::Vector3d &up)
+:
+initial_position_ { position }
+, initial_lookat_ { lookat }
+, initial_up_ { up }
+, context_ { context }
+, transform_ { look_at(initial_position_, initial_lookat_, initial_up_) }
+{
+    if (!context_)
+        throw std::runtime_error("Rendering context for Camera is nullptr!");
+
+    BOOST_LOG_TRIVIAL(debug) << transform_.matrix();
+
+    update_context(transform_);
+}
+
+
+// TODO: this method should be implemented inside RenderingContext itself, and be made thead safe (multiple cameras case)
+void Camera::update_context(const transform_type &t) const {
+    if (!context_)
+        throw std::runtime_error("Trying to access nullptr rendering context");
+
+    context_->update_transform(t);
+}
+
+math::Matrix4x4 Camera::look_at(const math::Vector3d& camera_position, const math::Vector3d& lookat, const math::Vector3d& up) {
+    position_ = camera_position;
+    direction_ = -((lookat - position_).normalized());
+    right_ = math::Vector3d ::cross(up, direction_).normalized();
+    up_ = math::Vector3d ::cross(direction_, right_);
+
+    auto left_part = math::Matrix4x4 {
+            right_.x(), right_.y(), right_.z(), 0,
+            up_.x(), up_.y(), up_.z(), 0,
+            direction_.x(), direction_.y(), direction_.z(), 0,
+            0, 0, 0, 1
+    };
+
+    auto right_part = math::Matrix4x4 {
+            1, 0, 0, -position_.x(),
+            0, 1, 0, -position_.y(),
+            0, 0, 1, -position_.z(),
+            0, 0, 0, 1
+    };
```

Comment:
```
Note that the rotation (left matrix) and translation (right matrix) parts are inverted (transposed and negated respectively) since we want to rotate and translate the world in the opposite direction of where we want the camera to move.
```